internal/reconciliations/istio/installation.go

```diff
@@ -33,27 +33,22 @@ const (
 
 // Reconcile runs Istio reconciliation to install, upgrade or uninstall Istio and returns the updated Istio CR.
 func (i *Installation) Reconcile(ctx context.Context, istioCR operatorv1alpha1.Istio, istioResourceListPath string) (operatorv1alpha1.Istio, error) {
-
 	istioTag := fmt.Sprintf("%s-%s", i.IstioVersion, i.IstioImageBase)
 
-	// We need to evaluate what changed since last reconciliation, to make sure we run Istio reconciliation only if it's necessary
-	istioCRChanges, err := EvaluateIstioCRChanges(istioCR, istioTag)
+	shouldInstallIstio, err := ShouldInstall(istioCR, istioTag)
 	if err != nil {
 		ctrl.Log.Error(err, "Error evaluating Istio CR changes")
 		return istioCR, err
 	}
 
-	if !istioCRChanges.requireIstioDeletion() && !hasInstallationFinalizer(istioCR) {
+	if !ShouldDelete(istioCR) && !hasInstallationFinalizer(istioCR) {
 		controllerutil.AddFinalizer(&istioCR, installationFinalizer)
 		if err := i.Client.Update(ctx, &istioCR); err != nil {
 			return istioCR, err
 		}
 	}
 
-	ctrl.Log.Info("Reconcile Istio installation", "Istio CR change evaluation", istioCRChanges)
-
-	if istioCRChanges.requireInstall() {
-
+	if shouldInstallIstio {
```

Comment:
I think we don't need this separation between Install and deletion. It might be confusing. What other cases than install and Uninstall are there? Since we only have deletion and "default handling" (call Istio install) at the moment. We can some conditions and make the code flow easier to follow.
So I think it might be more clear if we have something like
```
if shouldDelete
 Deletion logic
else
 Logic verifying Istio Version (currently in shouldInstall)
 Add finalizer if necessary
 Execute install logic
```

What do you think?

Reply:
So if we want to keep the shouldInstallIstio, then for me it makes more sense to have the initial positive action (the install) first and then the deletion as the else-if case. But I still don#t see the usage of an `else if` since we don't expect any different case.

```
if shouldInstall
  Add finalizer if needed
  Execute install logic
else if shouldDelete
  Deletion logics
```

Reply:
There is a different case - when delete action has been already called. In this case we won't execute install nor delete logic. It's also covered in tests.